qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -2,18 +2,19 @@
 from typing import cast, Dict, Union, Optional
 
 from qcodes import VisaInstrument, InstrumentChannel, ParameterWithSetpoints
-from qcodes.utils.validators import Enum, Numbers, Arrays, Ints
+from qcodes.utils.validators import Enum, Numbers, Arrays, Ints, Lists
 from qcodes.utils.helpers import create_on_off_val_mapping
 
 
-class NewParameterWithSetpoints(ParameterWithSetpoints):
+class ParameterWithSetpointsAllType(ParameterWithSetpoints):
     """
-    The newly added "_raw_value" will include extra fields for the data, in
-    addition to the numerical values, for the "sweep" parameter.
+    While the parent class ParameterWithSetpoints only support numerical data,
```

Comment:
> only support numerical data

this is not true. parameter with setpoints supports any kind of arrays, be it numeric, or string, or complex, etc. From how i understand the code, the docstring should say that `_raw_value` attribute is meant to be filled with some sort of "full data" while the parameters `get_raw`/`get_cmd` returns only a subset of that data, and it's up to the implementation of the `get_raw`/`get_cmd` to set `_raw_value` to this "full data" (i don't know how to write this better).

So, i think the `..AllType` suffix misses the point - and the point is to return on `.get()` the "important data" while on `get_selected` "more" data is accessible from that last `.get()` invocation. Could you try another iteration with the names? :)

also, it'd be great to have a better name for the `_raw_value`, raw value is something meaningful in the world of Parameters. Moreover, it's a bit unclear why get **selected** returns **raw** value, what does the **get** return then? :)

Reply:
But in the class definition of `ParameterWithSetpoints`, there is:
>         if not isinstance(vals, Arrays):
>           raise ValueError(f"A ParameterWithSetpoints must have an Arrays "
>                             f"validator got {type(vals)}")
where `Arrays` is "numerical numpy arrays of numeric types"

As for the naming game we are playing... how about `_raw_value` -> `_user_selected_data`, and the new class name `ParameterWithSetpoints**Customized**`?